src/detect-engine-analyzer.c

```diff
@@ -861,6 +861,14 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
                 jb_close(js);
                 break;
             }
+            case DETECT_ACK: {
+                const DetectAckData *cd = (const DetectAckData *)smd->ctx;
+
+                jb_open_object(js, "ack");
+                jb_set_uint(js, cd);
```

Comment:
I don't think this code would compile, not just because of the lack of the header inclusion that you have pointed out.

This function is missing an argument: it expects the JsonBuilder, a key, and an int value. (See bytetest, for example: https://github.com/OISF/suricata/blob/master/src/detect-engine-analyzer.c#L824 )

Check that example as well to get an idea on how to access the info you want to have logged out, as `cd` is the whole struct, and you want the inner `ack` value in it.